### ISSUE TITLE
Are you open to a pull request to pass secondary process arguments to the primary process?

### ISSUE THREAD

**Issue by sohailsomani** [itay-grudev/SingleApplication#10]
It would be great if the arguments to the secondary process were passed to the master process so if I launched the secondary process like: `MySingleApplication --arg1=value1 --arg2=value2`, then the `showUp()` signal is modified to pass those along as a `QStringList`: `showUp(QStringList const & args)`

I would do the work for this as I need functionality like this. I believe `QtSingleApplication` used to do this.

**Comment by itay-grudev** [comment]
I am planning to do the following:
* Add a single boolean argument that allows an instance to run as secondary
* Expose the `QLocalSocket` so you can send anything from a secondary instance to the primary one.
* Bridge events or expose the `QLocalServer` on the primary instance so that information can be received.
This would allow for a wider range of scenarios not just sending the arguments of the program.

Unfortunately at the moment I have very little time. I planned to work on this two weeks ago. If you are interested in implementing it I'd love to help, though my assistance would be rather limited.

**Comment by sohailsomani** [comment]
Do you think it is necessary to expose the `QLocalSocket`? To refresh your memory, `QtSingleApplication` works like this below. For my use case, I would have just serialized a `QStringList` in some way and send that over using the `sendMessage` signal.

```
    int main(int argc, char **argv)
    {
        QtSingleApplication app(argc, argv);
        if (app.isRunning())
            return !app.sendMessage(someDataString);
        MyMainWidget mmw;
        app.setActivationWindow(&mmw);
        mmw.show();
        return app.exec();
    }
```

**Comment by itay-grudev** [comment]
I agree that "sendMessage" is quite simple to use, but exposing the `QLocalSocket` would give you more control.

**Comment by itay-grudev** [comment]
One of the main things that will need to be changed is the fact that the library closes the socket after a successful connection has been made from both server and client-side. Also the integer argument for allowing secondary instances is not practical, so that will have to go.

**Comment by sohailsomani** [comment]
I agree it gives you more control. What kind of use cases do you foresee for the `QLocalSocket`?

**Comment by itay-grudev** [comment]
I spent some time thinking and I came to the conclusion that since there will be some internal communication the socket and the server won't be exposed. Instead I will have a `sendMessage()` method and a `receivedMessage()` signal. `sendMessage()` will probably support not only QString but also QBuffer and etc.

**Comment by sohailsomani** [comment]
Awesome. I'm trying to use `QtSingleApplication` with 5.6 right now but if that doesn't work, I'll give you a patch for this project.

**Comment by itay-grudev** [comment]
Thanks.

**Comment by sohailsomani** [comment]
Just wanted to leave a note that `QtSingleApplication` worked fine with Qt 5.6 so you could probably just lift the code from their project to yours.

**Comment by itay-grudev** [comment]
Just for the record I implemented `sendMessage()` support in the new `v3.0a`.
